### PR TITLE
feat: Add additional content to third-party API

### PR DESCRIPTION
**Pull request by jcbcapps**
# SC-3054

## Proposed changes
- Add CMS Documents to third-party API for authenticated users
- Add InternalNews and LandingPage articles to third-party API for authenticated users

---

## Reviewer notes
You will need to create some articles on your local machine with the categories `InternalNews` and `LandingPage`
You will also need to follow the instructions for [generating a JWT](https://github.com/USSF-ORBIT/ussf-portal/blob/main/docs/how-to/generate-jwt-token.md)
- I've update the Postman collection and posted it in Slack
- After generating your token, you will find the new endpoints under `USSF Third Party API` > `Auth` in the Postman collection

### Start the system
```sh
yarn services:up
yarn dev
cd ../ussf-portal-cms
yarn dev
```

Login to the portal http://localhost:3000

### Start storybook

```sh
yarn storybook
```

Login to storybook http://localhost:6006, though the command above should open it for you

---

### As the original developer, I have

- [ ] Met the acceptance criteria
- [ ] Created new stories in Storybook if applicable
- [ ] Created/modified automated unit tests in Jest
- [ ] Created/modified automated [E2E tests](https://github.com/USSF-ORBIT/ussf-portal)
- [ ] Followed guidelines for zero-downtime deploys, if applicable
- [ ] Use [ANDI](https://www.ssa.gov/accessibility/andi/help/install.html) to check for basic a11y issues

### As a reviewer, I have

Check out our [How to review a pull request](https://github.com/USSF-ORBIT/ussf-portal/blob/main/docs/how-to/review-pull-request.md) document.

---

### CODE REVIEW

**Comment by jbecker01** [comment]
Works like a charm! 👍

**Review by gidjin — COMMENTED** [review]
Looks great and everything works, even using the jwt service. I didn't mark it approved because of the discussion around using internal news that we should have.

Suggestion: Can you update the e2e tests for this as well? You might already be on this.

File: src/pages/api/thirdparty/dataSources/thirdPartyKeystone.ts [hunk 1]
```diff
@@ -21,6 +24,36 @@ class ThirdPartyKeystoneAPI extends RESTDataSource {
...
+
+  async getDocuments() {
+    return this.post(`/api/graphql`, {
+      body: {
+        query: print(GET_DOCUMENTS),
+      },
+    })
+  }
+
+  async getInternalNewsArticles(publishedDate: DateTime) {
```
[the remark this answers is not in the document]
Comment (jcbcapps) [reply src/pages/api/thirdparty/dataSources/thirdPartyKeystone.ts:36]: Yeah that's a good point. Happy to discuss that with the team

[the remark this answers is not in the document]
Comment (jcbcapps) [reply src/pages/api/thirdparty/dataSources/thirdPartyKeystone.ts]: I actually went ahead and made the change 👍

**Comment by jcbcapps** [comment]
> Suggestion: Can you update the e2e tests for this as well? You might already be on this.

@gidjin I'm not sure how I would create an e2e test for this? I am testing everything I added in `resolvers.test.ts`

**Comment by gidjin** [comment]
> > Suggestion: Can you update the e2e tests for this as well? You might already be on this.
> 
> @gidjin I'm not sure how I would create an e2e test for this? I am testing everything I added in `resolvers.test.ts`

I agree. For reference we discussed a potential e2e test but existing ones cover enough of what we want tested there and the unit tests here cover everything else. Hence I'm approving :)

**Pull request merged**